projects/canopy/src/lib/paginator/paginator.component.ts

```diff
@@ -0,0 +1,282 @@
+   */
+  showFirstLastButtons?: boolean;
+}
+
+/**
+ * Injection token for setting default paginator options.
+ */
+export const LG_PAGINATOR_DEFAULT_OPTIONS = new InjectionToken<PaginationOptions>(
+  'LG_PAGINATOR_DEFAULT_OPTIONS',
+);
+
+let nextId = 0;
+
+/**
+ * A pagination component used to control the number of items shown in a page context.
+ *
+ * @example
+ * <lg-paginator [length]="tableItems.length" (page)="handleSelectPage($event)"></lg-paginator>
+ */
+@Component({
+  selector: 'lg-paginator',
+  templateUrl: './paginator.component.html',
+  styleUrls: ['./paginator.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LgPaginatorComponent {
+  /**
+   * Zero based index for the currently selected page.
+   */
+  @Input()
+  get pageIndex(): number {
+    return this._pageIndex;
+  }
+  set pageIndex(index: number) {
+    this._pageIndex = index;
+    this.setState();
+  }
+  private _pageIndex = 0;
+
+  /**
+   * The number of items shown per page.
+   */
+  @Input()
+  get pageSize(): number {
+    return this._pageSize;
+  }
+  set pageSize(size: number) {
+    this._pageSize = size;
+    this.pageIndex = 0;
+    this.setState();
+  }
+  private _pageSize = 10;
+
+  /**
+   * The total length of items available.
+   */
+  @Input()
+  get length(): number {
+    return this._length;
+  }
+  set length(length: number) {
+    this._length = length;
+    this.pageIndex = 0;
+    this.setState();
+  }
+  private _length: number;
+
+  /**
+   * The list of available page sizes to select.
+   */
+  @Input() pageSizes = [10, 20, 50, 100];
+
+  /**
+   * Flag to determine if the page size selection field should be shown.
+   */
+  @Input() hidePageSize = false;
+
+  /**
+   * Flag to determine if the first and last page controls should be shown.
+   */
+  @Input() showFirstLastButtons = false;
+
+  /**
+   * Output that emits a pagination event when the selected page index changes.
+   */
+  @Output() readonly page = new EventEmitter<PaginationEvent>();
+
+  /**
+   * The id attribute of the component host element.
+   */
+  @HostBinding('id')
+  id = `lg-paginator-${nextId++}`;
+
+  /**
+   * The lg-paginator class name added to the host element.
+   */
+  @HostBinding('class.lg-paginator')
+  class = true;
+
+  currentPageStart = 1;
+  currentPageEnd = 1;
+
+  prevIcon = lgIconChevronLeft.name;
+  nextIcon = lgIconChevronRight.name;
+  firstIcon = lgIconGotoFirst.name;
+  lastIcon = lgIconGotoLast.name;
+
+  /**
+   * Creates the paginator from any defined default options.
+   *
+   * @param cdr injected change detector ref
+   * @param lgIconRegistry injected canopy icon registry
+   * @param defaults default paginator options
+   */
+  constructor(
+    private cdr: ChangeDetectorRef,
+    lgIconRegistry: LgIconRegistry,
+    @Optional() @Inject(LG_PAGINATOR_DEFAULT_OPTIONS) defaults?: PaginationOptions,
+  ) {
+    lgIconRegistry.registerIcons([
```

Comment:
We usually register the icons in the component module. I don't think it changes much but maybe for consistency I would move it.